scylla/src/transport/iterator.rs

```diff
@@ -239,20 +237,35 @@ impl RowIterator {
             .unwrap_or(&*config.execution_profile.retry_policy)
             .new_session();
 
+        // We have to extract partition key here, because here we can return Err in case it is returned
+        // we can't do it in the worker future). However, we can't preserve the extracted partition key
+        // until it's needed inside the worker future, because in the meantime `config` is moved,
+        // and this invalidates the PartitionKey (as it holds refs to prepared statement and values).
+        // As a result, we have to extract partition key twice: here and inside the worker future.
+        // This shouldn't make a noticeable difference though.
```

Comment:
I don't understand why we can't return Err inside the worker future. When you encounter an error, you should send it using the channel that returns the "proof" of value being sent.

Reply:
I reasoned that there is a semantic difference between returning an error _before_ the Iterator Worker is spawned and _after_. It seems that, however, there is no such difference, so I'll use the channel as you suggest.